vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go

```diff
@@ -632,3 +640,84 @@ func TestFilterVPAsIgnoreNamespaces(t *testing.T) {
 
 	assert.ElementsMatch(t, expectedResult, result)
 }
+
+func TestCanCleanupCheckpoints(t *testing.T) {
+	client := fake.NewSimpleClientset()
+
+	_, err := client.CoreV1().Namespaces().Create(context.TODO(), &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "testNamespace"}}, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	vpaBuilder := test.VerticalPodAutoscaler().WithContainer("container").WithNamespace("testNamespace").WithTargetRef(&autoscalingv1.CrossVersionObjectReference{
+		Kind:       kind,
+		Name:       name1,
+		APIVersion: apiVersion,
+	})
+
+	balanced := vpaBuilder.WithRecommender("balanced").WithName("balanced").Get()
+	performance := vpaBuilder.WithRecommender("performance").WithName("performance").Get()
+	savings := vpaBuilder.WithRecommender("savings").WithName("savings").Get()
+	defaultVpa := vpaBuilder.WithRecommender("default").WithName("default").Get()
+
+	vpas := []*vpa_types.VerticalPodAutoscaler{balanced, performance, savings, defaultVpa}
+	vpaLister := &test.VerticalPodAutoscalerListerMock{}
+	vpaLister.On("List").Return(vpas, nil)
+
+	checkpoints := &vpa_types.VerticalPodAutoscalerCheckpointList{
+		Items: []vpa_types.VerticalPodAutoscalerCheckpoint{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "testNamespace",
+					Name:      "nonExistentVPA",
+				},
+				Spec: vpa_types.VerticalPodAutoscalerCheckpointSpec{
+					VPAObjectName: "nonExistentVPA",
+				},
+			},
+		},
+	}
+
+	for _, vpa := range vpas {
+		checkpoints.Items = append(checkpoints.Items, vpa_types.VerticalPodAutoscalerCheckpoint{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: vpa.Namespace,
+				Name:      vpa.Name,
+			},
+			Spec: vpa_types.VerticalPodAutoscalerCheckpointSpec{
+				VPAObjectName: vpa.Name,
+			},
+		})
+	}
+
+	checkpointClient := &fakeautoscalingv1.FakeAutoscalingV1{Fake: &core.Fake{}}
+	checkpointClient.Fake.AddReactor("list", "verticalpodautoscalercheckpoints", func(action core.Action) (bool, runtime.Object, error) {
+		return true, checkpoints, nil
+	})
+
+	deletedCheckpoints := []string{}
+	checkpointClient.Fake.AddReactor("delete", "verticalpodautoscalercheckpoints", func(action core.Action) (bool, runtime.Object, error) {
+		deleteAction := action.(core.DeleteAction)
+		deletedCheckpoints = append(deletedCheckpoints, deleteAction.GetName())
+
+		return true, nil, nil
+	})
+
+	feeder := clusterStateFeeder{
+		coreClient:          client.CoreV1(),
+		vpaLister:           vpaLister,
+		vpaCheckpointClient: checkpointClient,
+		clusterState:        &model.ClusterState{},
+		recommenderName:     "default",
+	}
+
+	feeder.GarbageCollectCheckpoints()
+
+	if !assert.Contains(t, deletedCheckpoints, "nonExistentVPA") {
+		return
+	}
```

Comment:
This can just be a simple `assert`, no need for the `if not - return`

```suggestion
	assert.Contains(t, deletedCheckpoints, "nonExistentVPA")
```

Reply:
fixed